### PR TITLE
[AKS] `az aks nodepool update`: Add `--enable/disable-fips-image` flags for GA mutable fips

### PR DESCRIPTION
**Pull request by pineapplethevoyager**
**Related command**
az aks nodepool update --enable-fips-image --disable-fips-image

**Description**
This PR allows the CLI functionality for the GA AKS feature allowing fips mode to be enabled/disabled after nodepool creation using the cli "az aks nodepool update" with associated flags above.

**Testing Guide**
#! Make sure to use region where AKS RP functionality has been released
az aks nodepool add -g chrislopezrg --cluster-name chrislopezaks -n fipstest2
az aks nodepool update -g chrislopezrg --cluster-name chrislopezaks -n fipstest --enable-fips-image
az aks nodepool show -g chrislopezrg --cluster-name chrislopezaks -n fipstest

az aks nodepool add -g chrislopezrg --cluster-name chrislopezaks -n fipstest3 --enable-fips-image
az aks nodepool add -g chrislopezrg --cluster-name chrislopezaks -n fipstest3 --disable-fips-image
az aks nodepool show -g chrislopezrg --cluster-name chrislopezaks -n fipstest3

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by yonzhan** [comment]
@FumingZhang to help review

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py [hunk 1]
```diff
@@ -1070,25 +1070,38 @@ def get_enable_ultra_ssd(self) -> bool:
         # this parameter does not need validation
         return enable_ultra_ssd
 
+    # Mutable Fips now allows changes after create
     def get_enable_fips_image(self) -> bool:
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:1074]: looks like worth adding some ut

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py [hunk 2]
```diff
@@ -1070,25 +1070,38 @@ def get_enable_ultra_ssd(self) -> bool:
...
-            self.agentpool and
-            hasattr(self.agentpool, "enable_fips") and      # backward compatibility
-            self.agentpool.enable_fips is not None
-        ):
-            enable_fips_image = self.agentpool.enable_fips
+        # In create mode, try and read the property value corresponding to the parameter from the `agentpool` object
+        if self.decorator_mode == DecoratorMode.CREATE:
+            if (
+                self.agentpool and
+                hasattr(self.agentpool, "enable_fips") and      # backward compatibility
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:1085]: make "enable_fips" a const?

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py [hunk 3]
```diff
@@ -1070,25 +1070,38 @@ def get_enable_ultra_ssd(self) -> bool:
...
         enable_fips_image = self.raw_param.get("enable_fips_image", False)
-        # try to read the property value corresponding to the parameter from the `agentpool` object
-        if (
-            self.agentpool and
-            hasattr(self.agentpool, "enable_fips") and      # backward compatibility
-            self.agentpool.enable_fips is not None
-        ):
-            enable_fips_image = self.agentpool.enable_fips
+        # In create mode, try and read the property value corresponding to the parameter from the `agentpool` object
+        if self.decorator_mode == DecoratorMode.CREATE:
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:1082]: what if it's UPDATE?
I think the mutually exclude check should also run?

**Review by andyliuliming — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py (hunk 2 above)
Comment (pineapplethevoyager) [reply src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:1085]: This code section isn't changed, and the string is not reused won't fix

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py (hunk 3 above)
Comment (pineapplethevoyager) [reply src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:1082]: If update the flag is read directly (ie if --enable-fips-image exists the fips-mode will be set to true)
This existing code only applies to when DecoratorMode.Create to check if it has been set elsewhere ie MC create which isn't allowed for update

This code is already in preview and has been tested by many parties with no issue

File: src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py (hunk 1 above)
Comment (pineapplethevoyager) [reply src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py:1074]: tests have been added under "test_agentpool_decorator.py" and "test_aks_commands.py"

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**